### ISSUE TITLE
Unexpecte behavior on cancelled request

### ISSUE THREAD

**Issue by ghost** [ThreeMammals/Ocelot#1045]
## Expected Behavior
On a cancelled request I would expect to find a single logged item and have RequestCanceled as error code. Also I would expect to have no stack trace information given the context, just the error code and the request path.

## Actual Behavior
I have 2 identical log items and the error code is UnableToCompleteRequestError. Below is how a log item looks.

```
requestId: "0HLQT4TOSU6LR:00000001", previousRequestId: "no previous request id", message: "Error Code: UnableToCompleteRequestError Message: Error making http request, exception: System.Threading.Tasks.TaskCanceledException: The operation was canceled. ---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request. ---> System.Net.Sockets.SocketException: The I/O operation has been aborted because of either a thread exit or an application request
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.FillAsync()
   at System.Net.Http.HttpConnection.ReadNextResponseHeaderLineAsync(Boolean foldedHeadersAllowed)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithNtConnectionAuthAsync(HttpConnection connection, HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Ocelot.Requester.HttpClientHttpRequester.GetResponse(DownstreamContext context) errors found in ResponderMiddleware. Setting error response for request path:/shipmentsapi/data/configurationtree, request method: GET"

```
## Steps to Reproduce the Problem

  1. Configure a route to a long running uri.
  2. Start a request (via web browser or a tool like soapui).
  3. Cancel the request before it completes.

## Specifications

  - Version: 13.5.2
  - Platform: Windows

**Comment by catcherwong** [comment]
When the exception is `OperationCanceledException` the error code will be `RequestCanceled`.

https://github.com/ThreeMammals/Ocelot/blob/fbd9c01a425a0a3bda46717efda423883543fb76/src/Ocelot/Requester/HttpExeptionToErrorMapper.cs#L17-L32

`System.Threading.Tasks.TaskCanceledException` is a subclass of `OperationCanceledException` and it can not trigger the `RequestCanceledError`. 

But I think it should get the `RequestCanceled` error code as well.

**Comment by catcherwong** [comment]
I send a PR to fix this problem.

Here is the new log information.

```
warn: Ocelot.Responder.Middleware.ResponderMiddleware[0]
      requestId: 0HLR1IQO6KT5M:00000001, previousRequestId: no previous request id, message: Error Code: RequestCanceled Message: The operation was canceled. errors found in ResponderMiddleware. Setting error response for request path:/aa, request method: GET
```

**Comment by raman-m** [comment]
@catcherwong 
:ok: Your fix is here
https://github.com/ThreeMammals/Ocelot/blob/d6eefc899f9b2d37814335c7d0bccd5f7a830bd0/src/Ocelot/Requester/HttpExceptionToErrorMapper.cs#L40-L43
Develop branch: [Line 40](https://github.com/ThreeMammals/Ocelot/blob/develop/src/Ocelot/Requester/HttpExceptionToErrorMapper.cs#L40)
Commit: https://github.com/ThreeMammals/Ocelot/commit/2c97c5859d260847d2220657e8946484784e5ea7
Good!

**Comment by raman-m** [comment]
Implemented by #1051
